dallinger/experiment_server/experiment_server.py

```diff
@@ -1650,6 +1650,8 @@ def _worker_complete(participant_id):
     if event_type is None:
         return
 
+    # Currently we execute this function synchronously, regardless of the
```

Comment:
Perhaps you should also correct the comment above regarding the `async worker function` and why the qualifications are done synchronously?

Reply:
> we might need to have some larger refactoring or discussion about what should and shouldn't be handled by the async worker and why

Completely agree. While working on this I was reminded how bizarre, and increasingly, burdensome, the way we overload "worker events" is: sometimes (and originally) these were by definition MTurk event handlers. Now they're in some cases Frankenstein monsters. _New_ event types (ie `BotAssignmentSubmitted`) are arguably fine, but in particular the way we re-purposed the `AssignmentSubmitted` event was a big mistake, IMO.